main.js

```diff
@@ -66,200 +104,134 @@ function displayArticles() {
 
-  for (let i = 0; i < subcomponentArray.length; i++) {
-    let currentSubcomponent = subcomponentArray[i];
-    if (i === 0) {
-      urlComponent += currentSubcomponent;
-    } else {
-      urlComponent += ` AND ${currentSubcomponent}`;
-    }
+	const newsDeskFilters = document.getElementById('newsdesk-fieldset');
+	const newsDeskValues = valuesFromFieldset(newsDeskFilters);
+  
+  if (newsDeskValues) {
+    filterValues.push(`news_desk:(${newsDeskValues})`);
   }
 
-  return urlComponent;
-}
-
-function getLocationURLComponent(locationString) {
-  locationString.trim();
-  locationString = encodeURIComponent(`"${locationString}"`);
-  return `glocations:(${locationString})`;
-}
-
-function getMaterialTypeURLComponent(typesArray) {
-  let filterOptions = '';
-
-  for (let i = 0; i < typesArray.length; i++) {
-    let currentType = typesArray[i].value;
-    
-    if (i === 0) {
-      filterOptions += `"${currentType}"`;
-    } else {
-      filterOptions += ` "${currentType}"`;
-    }
+  const materialTypeFilters = document.getElementById('material-types-fieldset');
+  const materialTypeValues = valuesFromFieldset(materialTypeFilters);
+  
+  if (materialTypeValues) {
+    filterValues.push(`type_of_material:(${materialTypeValues})`);
   }
 
-  filterOptions = encodeURIComponent(filterOptions);
-  return `type_of_material:(${filterOptions})`;
-}
-
-function getNewsDeskURLComponent(newsDeskArray) {
-  let filterOptions = '';
-
-  for (let i = 0; i < newsDeskArray.length; i++) {
-    let currentNewsDesk = newsDeskArray[i].value;
-    
-    if (i === 0) {
-      filterOptions += `"${currentNewsDesk}"`;
-    } else {
-      filterOptions += ` "${currentNewsDesk}"`;
-    }
+  let location = document.getElementById('location-search').value.trim();
+  
+  if (location) {
+    location = encodeURIComponent(`"${location}"`);
+    filterValues.push(`glocations.contains:(${location})`);
   }
-
-  filterOptions = encodeURIComponent(filterOptions);
-  return `news_desk:(${filterOptions})`;
+	
+	return filterValues;
 }
 
-function handleToggleFiltersClick(event) {
-  event.preventDefault();
-
-  const button = event.target;
-  const filtersDiv = document.getElementById('filters');
+function toggleFilterMenuVisibility() {
+  const filtersDiv = document.getElementById('filters-container');
   
-  if (!filtersDiv.style.display) {
+  if (filtersDiv.style.display === '') {
     filtersDiv.style.display = 'grid';
-    button.textContent = 'Hide filters';
+    filtersButton.textContent = 'Hide filters';
   } else {
     filtersDiv.style.display = '';
-    button.textContent = 'Show filters';
+    filtersButton.textContent = 'Show filters';
   }
 }
 
-function handlePaginationClick(event) {
-  if (event.target.id === 'next-page-btn') {
-    resultsPage++;
-  } else {
-    resultsPage--;
-  }
-
-  fetchArticles().then(() => {
-    scroll(0, 0);
-    updateNavDisplay();
-    displayArticles();
-  });
-}
+function valuesFromFieldset(fieldset) {
```

Comment:
it is really good that you implemented this yourself. later you will probably use a library for this, but it is good now that you know how they work.